Control.cpp

```diff
@@ -45,61 +50,284 @@ double Control::getAngleError(const double current_angle_x, const double desire)
   return error;
 }
 
-double Control::getErrorUltrasonic(const double current_distance, const double desireUltrasonic) {
-  double error = current_distance - desireUltrasonic;
-  return error;
+double Control::getErrorUltrasonic(const double current_distance, const double desire_ultrasonic) {
+  return(current_distance - desire_ultrasonic);
 }
 
 void Control::getPwm(double &speed) { 
-  if (speed < kLimit_inf_pwm) {
-    speed = kLimit_inf_pwm;
+  if (speed < Common::kLimit_inf_pwm) {
+    speed = Common::kLimit_inf_pwm;
   }
   
-  else if (speed > kLimit_sup_pwm) {
-    speed = kLimit_sup_pwm;
+  else if (speed > Common::kLimit_sup_pwm) {
+    speed = Common::kLimit_sup_pwm;
   }
 }
 
 double Control::getNewDesireLeft(double new_desire) {
   new_desire = new_desire - kDegrees90;
   if (new_desire < 0) {
-    new_desire = new_desire + kDegrees360;
+    new_desire += kDegrees360;
   }
   return new_desire;
 }
 
 double Control::getNewDesireRight(double new_desire) {
-  new_desire = new_desire + kDegrees90;
+  new_desire += kDegrees90;
   if (new_desire > kDegrees360) {
-    new_desire = new_desire - kDegrees360;
+    new_desire -= kDegrees360;
   }
   return new_desire;
 }
 
 bool Control::detectRamp() {
-  return (bno_.getAngleY() < -kLimitInfDegrees 
-  && bno_.getAngleY() > -kLimitSupDegrees 
-  && bno_.getAngleZ() < kRangeAngleZ 
-  && bno_.getAngleZ() > -kRangeAngleZ);
+  return (bno_->getAngleY() < -kLimitInfDegrees 
+  && bno_->getAngleY() > -kLimitSupDegrees 
+  && bno_->getAngleZ() < kRangeAngleZ 
+  && bno_->getAngleZ() > -kRangeAngleZ);
 }
 
 bool Control::bumperLevel1() {
-  return (bno_.getAngleY() < -kLimitInfBumper1 
-  && bno_.getAngleY() > -kLimitSupBumper1 
-  && bno_.getAngleZ() > kRangeAngleZ 
-  || bno_.getAngleZ() < -kRangeAngleZ);
+  if (bno_->getAngleY() < -kLimitInfBumper1 && bno_->getAngleY() > -kLimitSupBumper1) {
+    if (bno_->getAngleZ() > kRangeAngleZ || bno_->getAngleZ() < -kRangeAngleZ) {
+      return true; // It's a bumper level 1.
+    }
+    else {
+      return false; // It's a ramp.
+    }
+  }
+  else {
+    return false; // It's another bumper level.
+  }
```

Comment:
The first way was almost correct. Let me explain:
if(A) {
  if(B) {
    return true;
  }
  else {
    return false;
  }
}
return false;

is the same as:
return A && B;
Now,
if A = A1 && A2
and B = B1 || B2
Then:
return (A1 && A2) && (B1 || B2);
And see how it is different than what you had:
return A1 && A2 && B1 || B2;